src/CMakeLists.txt

```
@@ -350,41 +350,7 @@ set_target_properties(Boost::MPL PROPERTIES
     "BOOST_MPL_CFG_NO_PREPROCESSED_HEADERS;BOOST_MPL_LIMIT_LIST_SIZE=30")
 
 if(WITH_SEASTAR)
-  find_package(c-ares 1.13.0 QUIET)
-  if(NOT c-ares_FOUND)
-    message(STATUS "Could not find c-ares, will build it")
-    include(Buildc-ares)
-    build_c_ares()
-  endif()
-  macro(find_package name)
-    if(NOT TARGET ${name})
-      _find_package(${ARGV})
-    endif()
-  endmacro ()
-  set(Seastar_API_LEVEL "6" CACHE STRING "" FORCE)
-  set(Seastar_HWLOC OFF CACHE BOOL "" FORCE)
-  set(Seastar_STD_OPTIONAL_VARIANT_STRINGVIEW ON CACHE BOOL "" FORCE)
-  if(Seastar_DPDK)
-    find_package(dpdk QUIET)
-    if(NOT DPDK_FOUND)
-      include(BuildDPDK)
-      build_dpdk(${CMAKE_BINARY_DIR}/src/dpdk)
-    endif()
-  endif()
-  list(APPEND Seastar_CXX_FLAGS
-    "-DSEASTAR_NO_EXCEPTION_HACK"
-    "-Wno-error"
-    "-Wno-sign-compare"
-    "-Wno-attributes"
-    "-Wno-pessimizing-move"
-    "-Wno-address-of-packed-member"
-    "-Wno-non-virtual-dtor")
-  set(Seastar_CXX_FLAGS "${Seastar_CXX_FLAGS}" CACHE STRING "" FORCE)
-  add_subdirectory(seastar)
-  # create the directory so cmake won't complain when looking at the imported
-  # target: Seastar exports this directory created at build-time
-  file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/seastar/gen/include")
-  add_subdirectory(crimson)
+  message(STATUS "WITH_CRIMSON can only be built with squid and above")
```

In reef the option name is `WITH_SEASTAR`, not `WITH_CRIMSON`:
```suggestion
  message(STATUS "Crimson can only be built with squid and above")
```

---

... and it seems like with https://github.com/ceph/ceph-build/pull/2346 in place this PR isn't strictly needed because `WITH_SEASTAR` should default to OFF and `WITH_CRIMSON` environment variable wouldn't be picked up by `run-make-check.sh` and `install-deps.sh` in reef?

---

> ... and it seems like with [ceph/ceph-build#2346](https://github.com/ceph/ceph-build/pull/2346) in place this PR isn't strictly needed because `WITH_SEASTAR` should default to OFF and `WITH_CRIMSON` environment variable wouldn't be picked up by `run-make-check.sh` and `install-deps.sh` in reef?

You're right, I didn't consider this. I'm glad that the change mentioned was merged prior to T dev freeze.
The above deprecation is also correct for Squid as we are not backporting any fixes to it as well.
So currently I don't see a reason to keep building in S as well.

Essentially this means that Crimson is only built (and supported)  in **main** (and in T once we freeze). 
Closing this one. Thanks for spotting this.